src/streamlink/plugin/plugin.py

```diff
@@ -323,5 +323,10 @@ def get_streams(self, *args, **kwargs):
     def _get_streams(self):
         raise NotImplementedError
 
+    def stream_title(self):
```

Comment:
The relation between `stream_title` and `_get_title` seems to be basically the same as `streams`|`get_streams` and `_get_streams`: `get_streams` is deprecated in favour of `streams` but all plugins still use `_get_streams`. Personally, I like the way this decouples what plugins do from what the public Plugin API does.

Either way, having `_get_title` call itself is obviously undesirable. You could simply remove `stream_title` and leave `_get_title` as is, and it would have the same effect without the recursion. Or perhaps:
```python
    def stream_title(self):
        return self._get_title()

    def _get_title(self):
        return self.url```
The help description says it already defaults to the url, but I don't see where that happens (in fact, it looks like the title is left as "Untitled").